tests/integration/awslambda/test_lambda.py

```diff
@@ -696,6 +696,22 @@ def test_url_config_lifecycle(self, lambda_client, create_lambda_function, snaps
             lambda_client.get_function_url_config(FunctionName=function_name)
         snapshot.match("failed_getter", ex.value.response)
 
+    def test_large_environment_variables(self, lambda_client, create_lambda_function, snapshot):
+        """Lambda functions with environment variables larger than 4 KiB should fail to create."""
+        snapshot.add_transformer(snapshot.transform.lambda_api())
+        large_envar = "x" * 5 * 1024
+        with pytest.raises(lambda_client.exceptions.InvalidParameterValueException) as ex:
+            create_lambda_function(
+                handler_file=TEST_LAMBDA_PYTHON_ECHO,
+                func_name="my-function",
+                runtime=LAMBDA_RUNTIME_PYTHON39,
+                envvars={
+                    "LARGE_VAR": large_envar,
+                },
+            )
+
+        snapshot.match("create_fn_result", ex.value.response)
```

Comment:
Another get-function would be nice here, to validate nothing was created even though the validation failed (which would be possible if the line `region.lambdas[arn] = lambda_function` was moved upwards a bit).